prov/util/src/util_main.c

```diff
@@ -131,7 +131,8 @@ static int util_find_domain(struct dlist_entry *item, const void *arg)
 
 	domain = container_of(item, struct util_domain, list_entry);
 
-	return !strcmp(domain->name, info->domain_attr->name) &&
+	return (!info->domain_attr->name ||
+		 !strcmp(domain->name, info->domain_attr->name)) &&
```

Comment:
Can you explain the issue more? I.e. if the info->domain_attr->name can be null, how are we not crashing?  I'm concerned that we're missing a test case here.

Also, we need the names to match.  If info->domain_attr->name is null, then the match should fail.

Reply:
It is indeed crashing, but only for providers that use util_getinfo() which invokes this util_find_domain() function. The only open-source provider that uses util_getinfo() is udp, which doesn't need RxM so you never hit that code path in existing tests.

My understanding (correct me please if I'm wrong) is that a null domain name means the caller doesn't care which provider domain it gets, rather than a declaration that it only wants a provider that has a null domain name.

Reply:
Okay, so at least the reason why we're not always crashing makes sense.  :)

IIRC, this call is used to find if there's an existing _opened_ domain that corresponds to the domain attributes being returned as part of a getinfo call.  Without a domain name, we don't know if we have the same domain or not.

I need to check that this assumption is right and look at the code flow. Maybe there's an issue in rxm not having the domain name set.

Reply:
Searching through the code, I see that util_getinfo() is called from the mlx, shm, tcp, and udp providers.  The util_getinfo() call obtains a list of fi_info structures from the calling provider.  That list should be fully formed, with all structures filled out appropriately.  A null domain name should not occur.

The search here is with the info list formed by the provider, not the hints that a user may have passed in.

So, I'm back to not seeing what the actual problem is.  If the domain name is null, then it seems that a provider is trying to return a domain without the name being set.

Reply:
Aha, you're right. This PR is based on my misunderstanding of the purpose of the domain_attr.name field, and my mis-analysis of the util_getinfo code path, so I'll close it. Sorry for the noise.